shelley/chain-and-ledger/executable-spec/test/Rules/TestDeleg.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Rules.TestDeleg where
+
+import           Control.Monad (when)
+import           Data.Map (Map)
+import qualified Data.Map.Strict as Map (lookup)
+import           Data.Set (Set)
+
+import           Hedgehog (Property, forAll, property, withTests, (===))
+
+import           Control.State.Transition.Generator (trace)
+import           Control.State.Transition.Trace (sourceSignalTargets, traceLength)
+
+import           Address (mkRwdAcnt)
+import           Coin (Coin)
+import           LedgerState (_delegationState, _delegations, _dstate, _pstate, _retiring, _rewards,
+                     _stKeys, _stPools)
+import           MockTypes (DELEG, DState, KeyHash, LedgerState, RewardAcnt, StakeCredential,
+                     StakePools)
+import           Slot (Epoch)
+import           TxData (pattern RegKey)
+
+import           Ledger.Core (dom, (∈))
+
+-------------------------------
+-- helper accessor functions --
+-------------------------------
+
+getStDelegs :: DState -> Set StakeCredential
+getStDelegs l = dom $ _stKeys l
+
+getRewards :: DState -> Map RewardAcnt Coin
+getRewards l = _rewards l
+
+getDelegations :: LedgerState -> Map StakeCredential KeyHash
+getDelegations l = _delegations $ _dstate $ _delegationState l
+
+getStPools :: LedgerState -> StakePools
+getStPools l = _stPools $ _pstate $ _delegationState l
+
+getRetiring :: LedgerState -> Map KeyHash Epoch
+getRetiring l = _retiring $ _pstate $ _delegationState l
+
+
+------------------------------
+-- Constants for Properties --
+------------------------------
+
+numberOfTests :: Int
+numberOfTests = 300
+
+numberOfSteps :: Int
+numberOfSteps = 100
```

Comment:
Maybe this can be called something like `traceLength`?

Reply:
Yeah, maybe someday we should rename `traceLength` in trace, and use `Trace.length` (importing `Trace` qualified).